src/index.js

```diff
@@ -1,30 +1,30 @@
 import { spawn } from 'child_process'
 import { resolve } from 'path'
+import { getParser } from './parsers'
 import 'core-js/shim'
-import failedSpecParser from './failed-spec-parser'
 import log from './logger'
 
 const DEFAULT_PROTRACTOR_ARGS = []
 
 const DEFAULT_OPTIONS = {
   nodeBin: 'node',
   maxAttempts: 3,
-  '--': DEFAULT_PROTRACTOR_ARGS,
-  protractorArgs: DEFAULT_PROTRACTOR_ARGS
+  protractorArgs: DEFAULT_PROTRACTOR_ARGS,
+  parser: false
 }
 
 export default function (options = {}, callback = function noop () {}) {
   let parsedOptions = Object.assign(DEFAULT_OPTIONS, options)
   let testAttempt = 1
-  // todo: remove this in the next major version
-  parsedOptions.protractorArgs = parsedOptions.protractorArgs.concat(parsedOptions['--'])
 
-  function handleTestEnd (status, output) {
+  function handleTestEnd (status, output = '') {
     if (status === 0) {
       callback(status)
     } else {
       if (++testAttempt <= parsedOptions.maxAttempts) {
-        let failedSpecs = failedSpecParser(output)
+        let parser = getParser(parsedOptions.parser, output)
+        log('info', `Using ${parser.name} to parse output`)
```

Comment:
Ah yep! I wonder if it would be best to just add a check to the logger that sees if there's a `\n` at the end or not, that way it would work seamlessly with stdout from protractor.